### PR TITLE
Fix bug flushing Data frames from asyncio client

### PR DESCRIPTION
**Pull request by coretl**
When AsyncioClient.data() was called with non-zero flush_period then it only yielded the Data after flush when a new packet of data appeared on the wire or at the end of acquisition. This changes it to be more eager and yield as soon as it is flushed.

### CODE REVIEW

**Comment by dmgav** [comment]
@coretl I just tested the code with PandABox and PandA IOC. Panda was configured to generate 12 trigger pulses, which resulted in 12 data points saved to HDF5 file. So this seems to fix the original issue. But now I see the following error message (IOC remains functional):

```
INFO:Arming PandA
INFO:Opened '/nsls2/users/dgavrilov/panda-test/panda_data_20230901_110824.h5' with 16 byte samples stored in 3 datasets
INFO:Requested number of frames (12) captured, disabling Capture.
INFO:Closed '/nsls2/users/dgavrilov/panda-test/panda_data_20230901_110824.h5' after writing 12 samples. End reason is 'Ok'
ERROR:_GatheringFuture exception was never retrieved
future: <_GatheringFuture finished exception=CancelledError()>
Traceback (most recent call last):
  File "/nsls2/users/dgavrilov/conda_envs/2023-2.1-py310-tiled-panda/lib/python3.10/asyncio/tasks.py", line 605, in sleep
    return await future
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

asyncio.exceptions.CancelledError
INFO:Disarming PandA
```

File: pandablocks/asyncio.py [hunk 1]
```diff
@@ -142,35 +141,41 @@ async def data(
...
-                for d in connection.receive_bytes(received, flush_every_frame):
-                    data.append(d)
-                while data:
-                    yield data.popleft()
+                for data in await queue.get():
+                    yield data
         finally:
-            flush_task.cancel()
-            await data_stream.close()
+            fut.cancel()
```
Comment (dmgav) [inline pandablocks/asyncio.py:180]: Adding the following code after this line seems to catch the exception and prevent traceback from being printed
```suggestion
            fut.cancel()
            try:                                                                                                                                                                
                await fut
            except asyncio.CancelledError:
                pass
```

Comment (dmgav) [reply pandablocks/asyncio.py:180]: May be there is a better way to capture the exception

Comment (coretl) [reply pandablocks/asyncio.py:181]: Thanks, I've applied your idea (but using `contextlib.suppress`)

**Comment by coretl** [comment]
@dmgav does this work for you now? Should I merge?

**Comment by dmgav** [comment]
Yes, position capture works now without issues.

**Pull request merged**